aws/resource_aws_globalaccelerator_endpoint_group.go

```diff
@@ -0,0 +1,360 @@
+				Optional: true,
+				Computed: true,
+			},
+			"health_check_path": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"health_check_port": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Computed: true,
+			},
+			"health_check_protocol": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					globalaccelerator.HealthCheckProtocolTcp,
+					globalaccelerator.HealthCheckProtocolHttp,
+					globalaccelerator.HealthCheckProtocolHttps,
+				}, false),
+			},
+			"threshold_count": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Computed: true,
+			},
+			"traffic_dial_percentage": {
+				Type:     schema.TypeFloat,
+				Optional: true,
+				Computed: true,
+			},
+			"endpoint_configurations": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 10,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"endpoint_id": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"weight": {
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceAwsGlobalAcceleratorEndpointGroupCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).globalacceleratorconn
+
+	opts := &globalaccelerator.CreateEndpointGroupInput{
+		ListenerArn:         aws.String(d.Get("listener_arn").(string)),
+		IdempotencyToken:    aws.String(resource.UniqueId()),
+		EndpointGroupRegion: aws.String(d.Get("endpoint_group_region").(string)),
+	}
+
+	if v, ok := d.GetOk("health_check_interval_seconds"); ok {
+		opts.HealthCheckIntervalSeconds = aws.Int64(int64(v.(int)))
+	}
+
+	if v, ok := d.GetOk("health_check_path"); ok {
+		opts.HealthCheckPath = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("health_check_port"); ok {
+		opts.HealthCheckPort = aws.Int64(int64(v.(int)))
+	}
+
+	if v, ok := d.GetOk("health_check_protocol"); ok {
+		opts.HealthCheckProtocol = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("threshold_count"); ok {
+		opts.ThresholdCount = aws.Int64(int64(v.(int)))
+	}
+
+	if v, ok := d.GetOk("traffic_dial_percentage"); ok {
+		opts.TrafficDialPercentage = aws.Float64(v.(float64))
+	}
+
+	if v, ok := d.GetOk("endpoint_configurations"); ok {
+		opts.EndpointConfigurations = resourceAwsGlobalAcceleratorEndpointGroupExpandEndpointConfigurations(v.([]interface{}))
+	}
+
+	log.Printf("[DEBUG] Create Global Accelerator endpoint group: %s", opts)
+
+	resp, err := conn.CreateEndpointGroup(opts)
+	if err != nil {
+		return fmt.Errorf("Error creating Global Accelerator endpoint group: %s", err)
+	}
+
+	d.SetId(*resp.EndpointGroup.EndpointGroupArn)
+
+	acceleratorArn, err := resourceAwsGlobalAcceleratorListenerParseAcceleratorArn(d.Id())
+	if err != nil {
+		return err
+	}
+
+	// Creating an endpoint group triggers the accelerator to change status to InPending
```

Comment:
This logic should reuse the available function (albeit slightly poor naming): `resourceAwsGlobalAcceleratorAcceleratorWaitForState`

```go
err = resourceAwsGlobalAcceleratorAcceleratorWaitForState(conn, acceleratorArn)
if err != nil {
	return err
}
```